src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldProjection.java

```diff
@@ -45,6 +47,9 @@
 public class NestedFieldProjection {
 
     private final SearchRequestBuilder request;
+    private static final String SEPARATOR = ".";
+    private String lastFieldInPath = "";
```

Comment:
I think we need to add more detailed comments when introducing state to a class. Please refer to https://github.com/opendistro-for-elasticsearch/sql/blob/master/src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/nestedfield/NestedFieldRewriter.java#L32.

Reply:
Added documentation with similar state transition table.